internal/hmyapi/apiv2/transactionpool.go

```diff
@@ -362,56 +362,42 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 }
 
 // PendingTransactions returns the plain transactions that are in the transaction pool
-// and have a from address that is one of the accounts this node manages.
 func (s *PublicTransactionPoolAPI) PendingTransactions() ([]*RPCTransaction, error) {
 	pending, err := s.b.GetPoolTransactions()
 	if err != nil {
 		return nil, err
 	}
-	managedAccounts := make(map[common.Address]struct{})
-	transactions := make([]*RPCTransaction, 0, len(pending))
-	for _, tx := range pending {
-		var signer types.Signer = types.HomesteadSigner{}
-		if tx.Protected() {
-			signer = types.NewEIP155Signer(tx.ChainID())
-		}
-		from, _ := types.PoolTransactionSender(signer, tx)
-		if _, exists := managedAccounts[from]; exists {
-			if plainTx, ok := tx.(*types.Transaction); ok {
-				transactions = append(transactions, newRPCPendingTransaction(plainTx))
-			} else if _, ok := tx.(*staking.StakingTransaction); ok {
-				continue // Do not return staking transactions here
-			} else {
-				return nil, types.ErrUnknownPoolTxType
+	transactions := make([]*RPCTransaction, len(pending))
+	for i := range pending {
+		if plainTx, ok := pending[i].(*types.Transaction); ok {
+			if tx := newRPCTransaction(plainTx, common.Hash{}, 0, 0, 0); tx != nil {
+				transactions[i] = tx
```

Comment:
did you try mixed transactions in the pending? index won't be correct, as you skip staking tx here?

Reply:
Right, will swap to list building.

Reply:
https://github.com/harmony-one/harmony/pull/2937